shell/edit/provisioning.cattle.io.cluster/rke2.vue

```diff
@@ -331,6 +331,26 @@ export default {
 
       const out = findBy(this.versionOptions, 'value', str);
 
+      // Adding the option 'none' to Container Network select (used it Basics component)
+      // https://github.com/rancher/dashboard/issues/10338
+      if (!out.serverArgs) {
+        out.serverArgs = {};
+      }
+
+      if (!out.serverArgs.cni) {
+        out.serverArgs.cni = {};
+      }
+
+      if (!out.serverArgs.cni?.options) {
+        out.serverArgs.cni.options = [];
```

Comment:
I think this will make showCni on L527 to be always true? I think we should either only set none if cni options already exist or remove showCni and its dependencies.

Reply:
Yep, I think you are right. Thanks for catching this!